commons/zenoh-protocol/src/scouting/hello.rs

```diff
@@ -129,3 +130,39 @@ impl Hello {
         }
     }
 }
+
+/// A zenoh Hello message.
+pub struct Hello(HelloProto);
```

Comment:
I would expect this type to be defined in the API and not in `zenoh-protocol`.

Reply:
On the one hand this is right, but sometimes it doesn't work. Especially this doesn't work for `ZenohId`: it's a wrapper for `ZenohIdProto` and logically should be in `zenoh`, but it's also used by `zenoh-config` so it should be outside of `zenoh` to avoid circular dependency.
So the approach when all wrappers for reexported structures should be located near the original structure is systematic and guarantees no dependency problems. Another solution would be to add special crate for wrapped structures which can be reused by other crates. But 1) in some rare cases this also can cause circular dependency and 2) this would require adding one more crate, increasing complexity of our project

So I think that the approach when reexported wrappers are declared near the original structures is the most acceptable

Reply:
I think the real problem is that `zenoh-config` uses *API-level* types directly from `zenoh-protocol`. But `zenoh-protocol` per-se does not need to care about those wrapper. 

In `zenoh-protocol` everything is defined and exported as public to detect as many breaking changes as possible throughout the codebase as soon as one minimal change happens. Adding the wrapper inside `zenoh-protocol` may lead to more misusage to me than anything else. Could the wrapper defined in `zenoh-config` and then re-exported by `zenoh` and then leave untouched the `zenoh-protocol`?

Reply:
Ok, that's another solution. Let's put all such wrappers to `zenoh-config`. This will cause reexport duplication at this moment because we reexporting `zenoh-config::*` in `config` crate and also should reexport `Hello` in `scoiting`, etc. But I think this is not a big issue at this moment, but better to make reexport from `zenoh-config` explicit in 1.0.0